database/db.js

```diff
@@ -13,22 +13,34 @@ const room = mongoose.Schema({
   date: Date,
 });
 
+const roomsSchema = mongoose.Schema({
+  roomName: String,
+  room: [bookingSchema],
+});
+
+const hotelSchema = mongoose.Schema({
+  id: Number,
+  rooms: [roomsSchema],
+});
 
-// in the future I will make a hotel schema with a hotelId containing a child schema called rooms
-// const hotelSchema = mongoose.Schema({
-//   id: Number,
-//   rooms: [room],
-// });
-const Booking = mongoose.model('Booking', room);
-// const Hotel = mongoose.model('Hotel', hotelSchema);
+const Booking = mongoose.model('Booking', bookingSchema);
+const Hotel = mongoose.model('Hotel', hotelSchema);
+const Room = mongoose.model('Room', roomsSchema);
 
 const save = (reservation) => {
-  const booking = new Booking(reservation);
-  booking.save();
+  reservation.save();
 };
-// in the future, if i use this function, this will save only the parent schema which
-// has the child schemas already pushed into it
-// const save = (reservation) => {
-  // reservation.save();
-// }
+
+const serveHotel = (callback) => {
+  Hotel.findOne({ id: 1 }).exec((err, data) => {
+    if (err) callback(err, null);
+    else callback(null, data);
+  });
+};
+
+
 module.exports.save = save;
+module.exports.Hotel = Hotel;
+module.exports.allRooms = Room;
+module.exports.Booking = Booking;
+module.exports.serveHotel = serveHotel;
```

Comment:
It would be cleaner if you only had a single `module.exports` assignment and used ES6 Object shorthand notation to list your exports.

```
module.exports = {
  save,
  Hotel,
  Room,
 // etc...
}